api/hooks/discordNotifications/DiscordNotification.js

```diff
@@ -24,9 +29,18 @@ class DiscordNotification {
     try {
       const discordChannel = await this.getDiscordChannel(enrichedOptions.server.config.discordNotificationConfig[notificationOptions.notificationType]);
       if (discordChannel) {
-        discordChannel.send(embedToSend);
+        await discordChannel.send(embedToSend);
       }
     } catch (error) {
+      try {
+        const owner = await User.findOne(enrichedOptions.server.owner);
+        if (owner.discordId) {
+          const discordUser = await this.getDiscordUser(owner.discordId);
+          discordUser.send(`There was an error sending a CSMM notification to your channel: ${error}`);
```

Comment:
Needs an `await` here, right?


This might get pretty spammy for users. Could be good, cause that'll force them to fix the issue. Could be bad, cause they might just get rid of CSMM.

Perhaps we can do something similar to logging hook failures?
- Set a field in redis (or DB) that a failure has occurred
- Display it in the front end with some additional info (eg: check permissions?)

Reply:
Eh fat fingered the approve button :p

Reply:
> Needs an `await` here, right?

yea

> This might get pretty spammy for users. Could be good, cause that'll force them to fix the issue. Could be bad, cause they might just get rid of CSMM.

Could only do it on the first issue and let them know that discord notifications for this job is now disabled and disable it? I was trying to find the flag? I'm also not super concerned if they decide to get rid of csmm are you?

> Perhaps we can do something similar to logging hook failures?

Was thinking it might be worth having an inbox (notifications(serverId, type, message) table) where we can shove things like "your server's been disabled due to too many timeouts"

Reply:
> I'm also not super concerned if they decide to get rid of csmm are you?
Not really, but better UX is always good :)

> Was thinking it might be worth having an inbox (notifications(serverId, type, message) table) where we can shove things like "your server's been disabled due to too many timeouts"

Yep indeed, having a solid system for this would be good! Gives us a good way to surface errors to users and will likely help debugging issues. Depending on a users privacy settings, sending a DM on Discord will not always work either